Maestro/build.cmd

```diff
@@ -13,13 +13,22 @@ if not exist "%VS140COMNTOOLS%VsMSBuildCmd.bat" (
   exit /b 1
 )
 
+set cachedNuGetDir=%LocalAppData%\NuGet
+set cachedNuGet=%cachedNuGetDir%\NuGet.exe
 set logFile="%~dp0build.log"
 set solutionPath="%~dp0Microsoft.DotNet.Maestro.sln"
 set buildArgs=/nologo /maxcpucount /nodeReuse:false /v:minimal /clp:Summary /flp:Verbosity=detailed;LogFile=%logFile%;Append %*
 
 REM Add msbuild to the path for the current process.
 call "%VS140COMNTOOLS%\VsMSBuildCmd.bat"
 
+echo Downloading NuGet.exe v3.4.4.
+if not exist %cachedNuGetDir% mkdir %cachedNuGetDir%
+@powershell -NoProfile -ExecutionPolicy unrestricted -Command "$ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest 'https://dist.nuget.org/win-x86-commandline/v3.4.4/NuGet.exe' -OutFile '%cachedNuGet%'"
```

Comment:
To ensure that the machine has a version of NuGet.exe in this location.  Otherwise it might have a newer version which could break us.

Upon further thought, this behavior could break other apps which are depending on the version of NuGet.exe which is already there, so I think it would be better to put it in `%LocalAppData%\NuGet\3.4.4` instead.  In that case, unconditionally downloading the file isn't necessary to ensure that v3.4.4 is the one we use.